File: NodeProperty.js

//name: "balance", type: "Currency", visibility: "private", default: "0"
export class NodeProperty{
    constructor(name, type, visibility, default_value){
        this.name = name;
        this.type = type;
        this.visibility = visibility;
        this.default_value = default_value;
    }
}

File: server/DiagramrCodeGenerator/Model/NodeProperty.cs
namespace DiagramrCodeGenerator.Model
{
    public class NodeProperty : ICodePart
    {
        public string Name { get; set; }
        public string Type { get; set; }
        public string Visibility { get; set; }
        public string DefaultValue { get; set; }

        public string ToCode()
        {
            return $"{Visibility} {Type} {Name} {{ get; set; }} {(string.IsNullOrEmpty(DefaultValue)?string.Empty:$"= {DefaultValue}")};";
        }
    }
}
